.github/workflows/ci.yml

```diff
@@ -9,6 +9,10 @@ on:
   pull_request:
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Just to make sure I understand, if we push a new commit on the same branch, this will cancel in progress workflows on the previous commit and then run the workflows on the new commit?

Reply:
yes sorry, should have commented on that.  that's exactly right.  it just saves CI time.  
more here: https://docs.github.com/en/enterprise-cloud@latest/actions/using-jobs/using-concurrency#overview